pkg/streamingpromql/optimize/ast/prune_toggles.go

```diff
@@ -31,10 +47,34 @@ func (p *PruneToggles) Name() string {
 }
 
 func (p *PruneToggles) Apply(ctx context.Context, expr parser.Expr) (parser.Expr, error) {
-	return p.mapper.Map(ctx, expr)
+	p.mapper = NewPruneTogglesMapper()
```

Comment:
`Apply` might be called by multiple queries simultaneously, so setting `p.mapper` like this isn't safe. 

Do we need to store the mapper on `PruneToggles`? It looks like it could be a local variable.